logging/resource.go

```diff
@@ -126,15 +126,23 @@ func detectCloudFunction() *mrpb.MonitoredResource {
 	}
 }
 
-func (r *resource) isCloudRun() bool {
+func (r *resource) isCloudRunService() bool {
 	config := r.attrs.EnvVar("K_CONFIGURATION")
 	// note that this envvar is also present in Cloud Function environments
 	service := r.attrs.EnvVar("K_SERVICE")
 	revision := r.attrs.EnvVar("K_REVISION")
 	return config != "" && service != "" && revision != ""
 }
 
-func detectCloudRunResource() *mrpb.MonitoredResource {
+func (r *resource) isCloudRunJob() bool {
+	job := r.attrs.EnvVar("CLOUD_RUN_JOB")
+	execution := r.attrs.EnvVar("CLOUD_RUN_EXECUTION")
+	taskIndex := r.attrs.EnvVar("CLOUD_RUN_TASK_INDEX")
+	taskAttempt := r.attrs.EnvVar("CLOUD_RUN_TASK_ATTEMPT")
+	return job != "" && execution != "" && taskIndex != "" && taskAttempt != ""
```

Comment:
nit: not too important, but it would be nice if we could short-circuit if the first one fails, instead of always loading each value